tests/antithesis/main.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"log"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/ava-labs/avalanchego/genesis"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/tests/antithesis"
+	"github.com/ava-labs/avalanchego/tests/fixture/tmpnet"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+const NumKeys = 5
+
+func main() {
+	c, err := antithesis.NewConfig(os.Args)
+	if err != nil {
+		log.Fatalf("invalid config: %s", err)
+	}
+
+	ctx := context.Background()
+	antithesis.AwaitHealthyNodes(ctx, c.URIs)
+
+	if len(c.ChainIDs) != 1 {
+		log.Fatalf("expected 1 chainID, saw %d", len(c.ChainIDs))
+	}
+	chainID, err := ids.FromString(c.ChainIDs[0])
+	if err != nil {
+		log.Fatalf("failed to parse chainID: %s", err)
+	}
+
+	genesisWorkload := &workload{
+		id:      0,
+		chainID: chainID,
+		key:     genesis.VMRQKey,
+		addrs:   set.Of(tmpnet.HardhatKey.Address()),
+		uris:    c.URIs,
+	}
+
+	workloads := make([]*workload, NumKeys)
+	workloads[0] = genesisWorkload
+
+	for i := 1; i < NumKeys; i++ {
+		key, err := secp256k1.NewPrivateKey()
+		if err != nil {
+			log.Fatalf("failed to generate key: %s", err)
+		}
+
+		// TODO(marun) Transfer funds to the new key
+
+		workloads[i] = &workload{
+			id:      i,
+			chainID: chainID,
+			key:     key,
+			addrs:   set.Of(key.Address()),
+			uris:    c.URIs,
+		}
+	}
+
+	for _, w := range workloads[1:] {
+		go w.run(ctx)
+	}
+	genesisWorkload.run(ctx)
```

Comment:
Is the only reason we do this to block while this is running? Feel like we could just use a wg here to do the same thing without having to do the `workloads[1:]` trick:

```
        wg := &sync.WaitGroup{}
	for _, w := range workloads {
		go w.run(ctx, wg)
                wg.Add(1)
	}
        wg.Wait()
```

Reply:
Ah didn't realize this is copied.